grafana/grafana
# loki: alerting: do not round timestamps to seconds

in the loki-alerting mode code, when processing the Loki database response, we round the timestamps to seconds.

this causes problem when the interval (step) is smaller than 1 second, because then... example:
let's say the interval is 500milliseconds

the timestamps returned from Loki are: 1, 1.5, 2, 2.5, 3
we build the grafana dataframe with these timestamps: 1, 1, 2, 2, 3

this is not what we want.

the code removes the rounding.

all the test snapshots had to change because suddenly the timestamps in the dataframes are not rounded.

i also added one more dataframe-test-snapshot to test 1millisecond steps.

how to test:
- enable the new loki backend mode
- go to explore
- (the is-backend-mode-filter is a little flaky so please switch the mode to INSTANT, then back to RANGE)
- enter a metric query, like `count_over_time({level="info"}[10s)`, you should see a graph
- now zoom into the graph so that the whole time-range is just around 5seconds (this will make the interval smaller than 1 second)
- refresh the graph
- verify that the requst went in backend-mode by checking it in the browser-dev-tools, it should have gone to `/api/ds/query` not to `/api/datasources/proxy/...`
- verify that there are datapoints which differ in less than 1 second, some ways to do this:
    - mouseover the points in the graph and check their timestamp
    - look in the table below the graph
    - check in query inspector
